### PR TITLE
fix: parsing of the cascading app config

### PR DESCRIPTION
**Pull request by kpanot**

### CODE REVIEW

**Review by vscaiceanu-1a — DISMISSED** [review]

File: apps/github-cascading-app/src/cascading/cascading-probot.ts [hunk 1]
```diff
@@ -51,19 +51,21 @@ export class CascadingProbot extends Cascading {
       }).catch(() => null)));
 
     const configFileValidResponses = configFileResponses
-      .filter((res) => res && Array.isArray(res.data) && res.data.length > 0);
+      .map((res) => res && !Array.isArray(res.data) && res.data || null)
```
Comment (pginoux-1A) [inline apps/github-cascading-app/src/cascading/cascading-probot.ts:54]: Just to be sure I got it right: octokit.repos.getContent returns an array if the path given is a directory, so the goal here is to filter those outs ?

File: apps/github-cascading-app/src/cascading/cascading-probot.ts [hunk 2]
```diff
@@ -51,19 +51,21 @@ export class CascadingProbot extends Cascading {
...
 
     if (configFileValidResponses.length > 1) {
-      this.logger.warn(`Several configuration files have been found (${configFileValidResponses.map((c) => c && Array.isArray(c.data) && c.data[0].path).join(', ')}). The files will be ignored.`);
+      this.logger.warn(`Several configuration files have been found (${configFileValidResponses.map((c) => c?.path).join(', ')}). The files will be ignored.`);
     } else if (!configFileResponse) {
       this.logger.warn('No remote Configuration found, the default configuration will be used');
     } else {
       try {
-        const parsedConfig = Array.isArray(configFileResponse.data) && configFileResponse.data[0].content && JSON.parse(configFileResponse.data[0].content);
+        const configFileResponseWithContent: { content?: string; encoding?: 'utf8' | 'utf-8' | 'base64' } = configFileResponse as unknown as any;
```
Comment (pginoux-1A) [inline apps/github-cascading-app/src/cascading/cascading-probot.ts:65]: What's the difference here between as unknown as any and just as any ?

File: apps/github-cascading-app/src/cascading/cascading-probot.ts (hunk 1 above)
Comment (kpanot) [reply apps/github-cascading-app/src/cascading/cascading-probot.ts:54]: yes, we keep only files

File: apps/github-cascading-app/src/cascading/cascading-probot.ts (hunk 2 above)
Comment (kpanot) [reply apps/github-cascading-app/src/cascading/cascading-probot.ts:65]: `as any` was rising an exception at transpile time so I added the `as unknown` to make it passed (not sure why the `as any` was rising an exception here :S)

File: apps/github-cascading-app/src/cascading/cascading-probot.ts (hunk 1 above)
Comment (pginoux-1A) [reply apps/github-cascading-app/src/cascading/cascading-probot.ts:54]: Should never happen since it's driven by CONFIGURATION_FILES right ?

Comment (kpanot) [reply apps/github-cascading-app/src/cascading/cascading-probot.ts:54]: Could happen if someone use named a folder with one of the file names listed :D.
And it is also a way to ensure typing without casting

**Review by matthieu-crouzet — APPROVED** [review]

File: .husky/commit-msg
Comment (cpaulve-1A) [inline .husky/commit-msg:1]: why?

**Review by cpaulve-1A — APPROVED** [review]

File: .husky/commit-msg
Comment (kpanot) [reply .husky/commit-msg:1]: Give execution right for Unix/Linux based OS

**Pull request merged**